File: specification/specs2/x_pack/watcher/execute_watch/ExecutionResultAction.ts
class ExecutionResultAction {
	id: string;
	type: ActionType;
	status: Status;
	email: EmailActionResult;
	index: IndexActionResult;
	webhook: WebhookActionResult;
	logging: LoggingActionResult;
	pagerduty: PagerDutyActionResult;
	hipchat: HipChatActionResult;
	slack: SlackActionResult;
	reason: string;
}

File: output/csharp/XPack/Watcher/ExecuteWatch/ExecutionResultAction.cs
using Nest.XPack;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class ExecutionResultAction  {
		
		[DataMember(Name="email")]
		public EmailActionResult Email { get; set; }


		[DataMember(Name="id")]
		public string Id { get; set; }


		[DataMember(Name="index")]
		public IndexActionResult Index { get; set; }


		[DataMember(Name="logging")]
		public LoggingActionResult Logging { get; set; }


		[DataMember(Name="pagerduty")]
		public PagerDutyActionResult Pagerduty { get; set; }


		[DataMember(Name="reason")]
		public string Reason { get; set; }


		[DataMember(Name="slack")]
		public SlackActionResult Slack { get; set; }


		[DataMember(Name="status")]
		public Status Status { get; set; }


		[DataMember(Name="type")]
		public ActionType Type { get; set; }


		[DataMember(Name="webhook")]
		public WebhookActionResult Webhook { get; set; }

	}
}
